tests/linalg_test.py

```diff
@@ -1620,6 +1620,51 @@ def testToeplitzConstructionWithKnownCases(self):
       [3, 2, 1, 4]], dtype=np.float32))
 
 
+  base_shapes_small = [1, 2, 3]
+  base_shapes_big = [10, 100]
+  null_space_shapes = list(itertools.chain.from_iterable([[
+    (2, base_shape),
+    (base_shape, 2),
+    (1 + base_shape // 2, base_shape),
+  ] for base_shape in base_shapes_small + base_shapes_big]))
+  @jtu.sample_product(
+    shape=null_space_shapes,
+    dtype=float_types + complex_types)
+  def testNullSpace(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    args_maker = lambda: [X]
+    self._CheckAgainstNumpy(jtu.promote_like_jnp(osp.linalg.null_space),
+                            jsp.linalg.null_space, args_maker)
+
+  @jtu.sample_product(
+    shape=null_space_shapes,
+    dtype=float_types + complex_types,
+    size=[None, 2, 9])
+  def testNullSpaceCompiled(self, shape, dtype, size):
+    # test compiled version
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    if size != None:
+      jax.jit(jsp.linalg.null_space, static_argnames=('size'))(X, size=size)
```

Comment:
We should test that the output is as expected.

One suggestion along these lines: use `CheckAgainstNumpy` but define a scipy function wrapper that returns the expected padding. For example, here's how we test `jnp.nonzero` against numpy with the `size` argument: https://github.com/google/jax/blob/7e001d842e1cbfdbf991a4bd9b236012cc40fba4/tests/lax_numpy_test.py#L308-L315

Reply:
ok I see will adapt my test to have something exactly like this test

Reply:
done